bp_common/src/include/bp_common_me_if.vh

```diff
@@ -73,10 +78,16 @@
                                                                                                          \
   typedef struct packed                                                                                  \
   {                                                                                                      \
-    logic [data_width_mp-1:0]                    data;                                                   \
+    logic [lce_req_max_data_width_mp-1:0]        data;                                                   \
     bp_lce_cce_req_header_s                      header;                                                 \
   } bp_lce_cce_req_s;                                                                                    \
                                                                                                          \
+  typedef struct packed                                                                                  \
+  {                                                                                                      \
+    logic [cce_block_width_mp-1:0]               data;                                                   \
+    bp_lce_cce_req_header_s                      header;                                                 \
+  } bp_lce_cce_block_req_s;                                                                              \
```

Comment:
There's an interesting question here.  These interface structures are really "protocols".  And the protocol data width will be maximum 512 bits, based on the size field.

When we implement streaming mem_cmd/mem_resp interfaces, rather than block_width or req_width, the data parameter will refer to how wide the data bus width is (i.e. noc width, cache width).  In that case, having multiple structs isn't really necessary.  We're really just saying -- for this instantiation of the interface, we only expect data less than or equal to this size.

When you change the interface of the A better approach to removing extra buffering is to move that parameter to the wormhole adapters themselves.  For instance, by changing bp_me/src/v/wormhole/bp_me_wormhole_packet_encode_lce_req.v, this actually doesn't change the wormhole adapter buffer size at all, because that's based on len * protocol data width.

So I would say just combine bp_lce_cce_req_s and bp_lce_cce_block_req_s; we can look at the overhead in the backend, but most likely the vast majority of that will be better solved when we tackle serialization / deserialization

Reply:
Let me take another pass at it. I added the second struct because bp_tile instantiates both the LCEs and CCEs, and until we have the serdes changes, I figured we didn't want the D$ and I$ LCEs to have block_width data with the LCE Req network. I have a new idea for this though, which is to use the header struct already defined and concat it with a dword_width data signal. That should take care of routing a narrow signal from LCEs to adapters for now.

Agreed, these issues disappear with removing serialization/deserialization. That should be next step (or maybe just rolled in with this one)

Reply:
I like the header struct + data solution for the CCE, since that's (similar to) what the serialization approach will look like.  Maybe we can also use that in the LCE?

I'd suggest  getting rid of the lce_req_max_width_p parameter in the interface definition completely, (always CCE block width).  The extra wires anywhere the full packet is instantiated will get synthesized out, and there are no actual excess registers here (which would also get synthesized out)

We should definitely merge this before approaching the serialization project, which will take a few weeks at least.